### PR TITLE
Rely on official types

### LINKED ISSUE #44: Use definitely typed typings for stripe
Something to consider is adding more complete typing to the stripe instance. For example, I'm using
```typescript
(stripe as any).redirectToCheckout({
  sessionId: response.id
});
```
What about using the [definitelytyped typings](https://www.npmjs.com/package/@types/stripe-v3)?

_Originally posted by @yringler in https://github.com/AckerApple/stripe-angular/issues/42#issuecomment-752972863_

I created a branch in my fork, I deleted some of the types, switched to defnitely, and got build and tests to work.

Will work on deleting the rest of the types later.
One issue I had was some things are typed as Error, when really it should be a stripe.Error.
I'm assuming that it was just typed as Error for convenaince, and to get that one shared message property, and will liberaly change to stripe.Error.

IDK how that works with the exported library? Will it have the stripe types also? Or will user have to install stripe types?

That should be documented. Maybe we should have it as a peer dependency?

Currently in the package.json I added the types as a devDependency.

### PR DESCRIPTION
**Pull request by yringler**
Fixes #44

### CODE REVIEW

**Comment by yringler** [comment]
This is a huge PR in terms of effected changes, but not in terms of complexity.
I think main review should be in builds and tests passing.

File: dist/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "stripe-angular",
-  "version": "1.4.2",
+  "version": "1.4.3",
```
Comment (AckerApple) [inline dist/package.json:3]: Make this 1.5.0 please

**Comment by AckerApple** [comment]
Looking pretty good. I am on PTO so no action from me until next week

**Comment by yringler** [comment]
Updated version, and removed rest of custom types.
I ran into one issue: in [/app/src/components/app.component.ts](https://github.com/yringler/stripe-angular/blob/9dea81f6e4623b1d498d7eed9a05f59106289941/app/src/components/app.component.ts#L115), the official types are missing the metadata property, so I had to cast it... I know there's a way to extend interfaces etc, I gave it a shot, but I'm not so good at that... I opened a PR DefinitelyTyped/DefinitelyTyped/pull/50386 , will see what happens.
In the meantime I added a cast to this code

**Comment by AckerApple** [comment]
Just a note I am actively working on this project today so any notes, thoughts, and changes are best expressed today.

I will review and merge today

**Comment by yringler** [comment]
One note - I think this is built on an older version of angular? I know to get a part of this to build I had to specify the typescript version in package.json, because the current LTS was too new.
It would be nice if everything was updated.

**Pull request merged**